src/navigation/water_bottle_search.py

```diff
@@ -0,0 +1,426 @@
+from __future__ import annotations
+
+import heapq
+import random
+from threading import Lock
+from typing import Optional
+from scipy.signal import convolve2d
+
+
+import numpy as np
+
+import rospy
+from geometry_msgs.msg import Pose, PoseStamped, Point, Quaternion
+from std_msgs.msg import Header
+from mrover.msg import GPSPointList
+from nav_msgs.msg import OccupancyGrid, Path
+from navigation import approach_object, recovery, waypoint
+from navigation.context import convert_cartesian_to_gps, Context
+from navigation.trajectory import Trajectory, SearchTrajectory
+from util.SE3 import SE3
+from util.ros_utils import get_rosparam
+from util.state_lib.state import State
+
```

Comment:
I think you should probably abstract all AStar functionality into a class (maybe put it in another file too?) Just so this file isn't so huge and doing so many things